src/main/java/de/dennisguse/opentracks/content/data/Layout.java

```diff
@@ -1,16 +1,61 @@
 package de.dennisguse.opentracks.content.data;
 
+import android.content.res.Resources;
+import android.os.Parcel;
+import android.os.Parcelable;
+
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.List;
+import java.util.stream.Collectors;
+
+import de.dennisguse.opentracks.R;
+import de.dennisguse.opentracks.settings.PreferencesUtils;
+import de.dennisguse.opentracks.util.CsvConstants;
 
-public class Layout {
+public class Layout implements Parcelable {
     private final String profile;
+    private int columnsPerRow;
     private final List<DataField> dataFields = new ArrayList<>();
 
+    public static Layout fromCsv(String csvLine, Resources resources) {
+        List<String> csvParts = Arrays.asList(csvLine.split(CsvConstants.ITEM_SEPARATOR));
+        Layout layout = new Layout(csvParts.get(0), Integer.parseInt(csvParts.get(1)));
```

Comment:
Index out of bounds check.

Reply:
That would be a developer error. How could we handle this kind of mistakes? The error would come from another part and we should launch an Exception or something like this, right? Malformed CSV Layout or something like this.

Reply:
What about returning the default layout and print a log message with severity error?